### PR TITLE
Add POLICY.md

### PR DESCRIPTION
**Pull request by confused-Techie**
This PR adds a Policy file for our org.

The processes documented here are the ones that have naturally evolved from our usage and community, but deserve to be officially stated and documented.

### CODE REVIEW

File: POLICY.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Policy
+
+This document will lay out the official policies used within the Pulsar Edit Project. Documenting the policies that have come out of our normal usage, that deserve to be properly documented and viewable by anyone.
+
+## Polls
+
+A cornerstone of the way Pulsar interacts with the community is via Polls. These polls are usually created on Discord, but may in the future be expanded to other platforms as needed.
```
Comment (Daeraxa) [inline POLICY.md:7]:
```suggestion
A cornerstone of the way the Pulsar team interacts with the community is via Polls. These polls are usually created on Discord, but may in the future be expanded to other platforms as needed.
```

File: POLICY.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+A Poll is used to allow the community at large to easily and quickly contribute to a conversation and provide thier opinions on future development or goals of the Pulsar Project.
```
Comment (Daeraxa) [inline POLICY.md:9]:
```suggestion
A Poll is used to allow the community at large to easily and quickly contribute to a conversation and provide their opinions on future development or goals of the Pulsar Project.
```

File: POLICY.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+* Whenever a decision will effect a large percentage of users/contributors, a change is substantal enough, or will drastically effect a small percentage of users/contributors then a poll must be made.
```
Comment (Daeraxa) [inline POLICY.md:11]:
```suggestion
* Whenever we need to make a decision that will affect a large percentage of users/contributors, make a substantial enough change, drastically affect a small percentage of users/contributors then a poll must be made.
```

File: POLICY.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+* A Poll must be available for voting for at least 24 hours, to accommodate different timezones, and work schedules. Polls can be longer than this, but must never be shorter.
```
Comment (Daeraxa) [inline POLICY.md:12]:
```suggestion
* A Poll must be available for voting for at least 24 hours to accommodate different timezones and work schedules. Polls may be longer than this, but must never be shorter.
```

File: POLICY.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+* If a poll fails, then you must not recreate the exact same poll right afterwards. If the same idea has new merit or new validation at a later time, communicate with the community to determine if another poll is reasonable.
+
+## Repeat Questions
+
+Many times Pulsar throughout our various social platforms will receive repeated suggestions, and sometimes these suggestions are untenable for whatever various reasons. In these cases if we find ourselves repeatedly using the same answer, a discussion will happen at the Pulsar Admin Team level, on whether or not to add this conversation to our official [FAQ](https://github.com/pulsar-edit/.github/blob/main/FAQ.md) at which point, if agreed a PR will be made, and the community at large will be able to discuss it being added. At which point if it's added that now becomes the official stance and will be used in further discussion on the topic.
```
Comment (Daeraxa) [inline POLICY.md:17]:
```suggestion
Many times Pulsar throughout our various social platforms will receive repeated suggestions, and sometimes these suggestions are untenable for whatever reasons. In these cases if we find ourselves repeatedly using the same answer, a discussion will happen at the Pulsar Admin Team level on whether or not to add this conversation to our official [FAQ](https://github.com/pulsar-edit/.github/blob/main/FAQ.md) at which point, if agreed, a PR will be made and the community will be able to discuss it being added. The wording in that PR, if merged, then becomes the official stance on the topic and will be used in any further discussions.
```

File: POLICY.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+Obviously Pull Requests are an important part to software development.
+
+We get a lot of them, and there's not always enough of us to review everything in a timely manner. But below are some general guidelines.
+
+* Prior to merging there should be a best effort to allow two approvals of a PR.
+* A Pull Request should not be merged right away unless there is aduquate reason to do so. (e.g. This PR fixes a serious bug in the application, and is of a rather small change. This PR fixes a security issue.)
+
+## Donations
+
+We accept and encourage contributors to donate to the platform when they fiscally can. As these donations help Pulsar be great, and do what it needs to do.
```
Comment (Daeraxa) [inline POLICY.md:30]:
```suggestion
We accept and encourage all donations to the project. These donations help Pulsar be great, and to achieve our goals.
```
Felt a bit... like we were expecting people to donate to me

File: POLICY.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+A donation through Pulsar will always be used for the greater good of the editor and it's ecosystem at large. But donators must understand that the use of funds will be dependent on the discretion of Pulsars Core Maintainers.
```
Comment (Daeraxa) [inline POLICY.md:32]:
```suggestion
A donation to the Pulsar project will always be used for the greater good of the editor and its ecosystem. However, donators must understand that the use of funds will be left to the discretion of the Pulsar team.
```

File: POLICY.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+But officially when you donate this money is now used for the Pulsar project. A donation, unless as part of a gift for donating or other specifically indicated event, does not allow a donator any extra say in how Pulsar will operate. A donation does not allow directed use of the funds, and does not allow control of the funds after they have been donated.
```
Comment (Daeraxa) [inline POLICY.md:34]:
```suggestion
Any money donated can and will be used used for the Pulsar project. A donation, unless as part of a gift for donating or other specifically indicated event, does not allow a donator any extra say in how the Pulsar team will operate. A donation does not allow directed use of the funds, nore does it allow control of the funds after they have been donated.
```

> A donation, unless as part of a gift for donating or other specifically indicated event,

Not entirely sure what this means?

File: POLICY.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+Quickly put:
+
+When you donate to Pulsar, we are __not__ promising to spend money on anything in particular, nor will we necessarily be spending funds on anything other than our basic operating costs. Ultimately, maintianers maintain discretion to use the funds in the best interests of the Pulsar Project, and any decisions are ultimately done by maintianers in their capacity as folks with the responsibility over the project.
```
Comment (Daeraxa) [inline POLICY.md:38]:
```suggestion
When you donate to Pulsar, we are __not__ promising to spend money on anything in particular, nor will we necessarily be spending funds on anything other than our basic operating costs. Ultimately, the Pulsar team reserves discretion to use the funds in the best interests of the Pulsar Project, and any decisions are ultimately done by the team in their capacity as folks with the responsibility over the project.
```

File: POLICY.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+We thank you for any understanding, and sincerely appreciate, and express gratitude to anyone who donates.
```
Comment (Daeraxa) [inline POLICY.md:40]:
```suggestion
We thank you for any understanding, and we sincerely appreciate and express gratitude to anyone who donates.
```

**Review by Daeraxa — COMMENTED** [review]
Just my usual tweaks. Mostly just some rewording without changing the actual subject for readability or making some terms a little less ambiguous like referring to the Pulsar team vs maintainers.

File: POLICY.md (hunk 8 above)
Comment (confused-Techie) [reply POLICY.md:34]: That's just there, since in the future if we decide donating a certain value gets you a free sticker or shirt or something.

Basically a caveat to an event like that which will be super clear on OC's end.

File: POLICY.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
+# Policy
+
+This document will lay out the official policies used within the Pulsar Edit Project. Documenting the policies that have come out of our normal usage, that deserve to be properly documented and viewable by anyone.
```
[the remark this answers is not in the document]
Comment (confused-Techie) [reply POLICY.md:3]: This is a good point. Project makes sense for everything surrounding us, but Pulsar-Edit does make sense to.

Probably should go ahead and change it though

**Comment by confused-Techie** [comment]
@Daeraxa added the requested changes

File: POLICY.md (hunk 8 above)
Comment (Daeraxa) [reply POLICY.md]: Maybe "reward" as a word then? Basically unless we specifically said you get x reward for donating y... everything applies as usual.

**Comment by Daeraxa** [comment]
I made a comment on the "gift" one - not sure if it still shows after it becomes outdated

File: POLICY.md (hunk 8 above)
Comment (confused-Techie) [reply POLICY.md]: Yeah fair enough, I'll add that bit

**Comment by confused-Techie** [comment]
@Daeraxa It did not, but fixed it now with your suggestion

File: POLICY.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+* Prior to merging there should be a best effort to allow two approvals of a PR.
+* A Pull Request should not be merged right away unless there is aduquate reason to do so. (e.g. This PR fixes a serious bug in the application, and is of a rather small change. This PR fixes a security issue.)
+
+## Donations
+
+We accept and encourage all donations to the project. These donations help Pulsar be great, and to achieve our goals.
+
+A donation to Pulsar-Edit will always be used for the greater good of the editor and its ecosystem. However, donators must understand that the use of funds will be left to the discretion of the Pulsar team.
+
+Any money donated can and will be used used for Pulsar-Edit. A donation, unless as part of a reward for donating or other specifically indicated event, does not allow a donator any extra say in how the Pulsar team will operate. A donation does not allow directed use of the funds, nore does it allow control of the funds after they have been donated.
```
Comment (Daeraxa) [inline POLICY.md:34]:
```suggestion
Any money donated can and will be used used for Pulsar-Edit. A donation, unless as part of a reward for donating or other specifically indicated event, does not allow a donator any extra say in how the Pulsar team will operate. A donation does not allow directed use of the funds, nor does it allow control of the funds after they have been donated.
```
Fixing my own typo...

File: POLICY.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+## Polls
+
+A cornerstone of the way the Pulsar team interacts with the community is via Polls. These polls are usually created on Discord, but may in the future be expanded to other platforms as needed.
+
+A Poll is used to allow the community at large to easily and quickly contribute to a conversation and provide their opinions on future development or goals of Pulsar-Edit.
+
+* Whenever we need to make a decision that will affect a large percentage of users/contributors, make a substantial enough change, drastically affect a small percentage of users/contributors then a poll must be made.
+* A Poll must be available for voting for at least 24 hours to accommodate different timezones and work schedules. Polls may be longer than this, but must never be shorter.
+* If a poll fails, then you must not recreate the exact same poll right afterwards. If the same idea has new merit or new validation at a later time, communicate with the community to determine if another poll is reasonable.
```
Comment (DeeDeeG) [inline POLICY.md:13]: LONG VERSION

<details><summary>(personal philosophical rambling, skip to short version if need be, click to expand):</summary>

I know we _do_ polls.

I dunno about the "must" wording here, since we are not utterly consistent about it, and sometimes a PR needs to go out, say for security reasons for example, or there _is no consensus_ and yet somebody needs to pick something or move something forward.

Like, polls are IMO more for when we're not already sure as an admin team. We don't ask for input when there's already an obvious answer, even when that obvious answer affects a lot of people or some important constituency group.

So I think in all honestly the polls have been, and should be, a function of when the admins want/need help making the right call. If we require it based on who or how many people it affects, that may not align with what is practical and what in all honestly drives current poll behavior (in my personal view and observations).

Also: When this says "must" -- who is enforcing this, and what is the enforcement mechanism? If it's not a "must", or if it's not enforced, then this strikes me as more of a guideline, and like with IETF RFC's, I think these should be "should" rather than "must", and clarify that this is a guideline and documentation of an ongoing practice, moreso than a rule.

While community is a major, major aligning principle of the project, community members who aren't mods are neither as engaged in every little decision we might have to make, nor IMO could they or should they be ultimately responsible over everything, there has to be an admin team for things to land on good and bad, and this starts to feel overly prescriptive as written in my personal view. Though we do obviously take community perspective on things pretty often. But it's often in terms of who pops into issues and who is in the Discord server, not _necessarily_ a poll. Heck, I hardly consider the admin team much distinct from the rest of the community other than I know they will show up, and they have the responsibility associated with the level of access they are granted. Other than that we _are_ community members. And the informal nature of our association is shown by how many decisions we tend to make without polls, there is, thus far, a trust that is shared, IMO, to move in the right direction and everybody pitch in as and when needed.

I think implicitly a lot of the "community" vibe of Pulsar is that there is an _existing_ community of Atom users who would be sad to see an editor like Atom go without a viable similar replacement, and that by respecting the wish of many people to have a viable and similar replacement, we are reaching to meet the (often unspoken or informally organized) demand for such things to be available.

So community is a principle, but I don't think you can prescribe respect for community as a rule, it's ethos and it manifests in how we do everything, but it still is a project that needs a leadership team and executive decisions to be made, etc. So we can't mandate polls and have it be smooth and well functioning, or at least I am skeptical of the implication that we _already_ do that and skeptical of the requirement that we should _start_ doing that.

</details>

---

SHORT VERSION:

Okay, so, getting off my soap box.

I would suggest changing "must" in this passage to "should", or even consider just describing this as "a thing we do" and not prescribe it or suggest it is necessary in any way. Just that it's a convention and we prefer it in a lot of situations where we need community input.

Like if we drift to not ever doing polls, something would have gone "off"/wrong compared to how this document describes things.

Comment (confused-Techie) [reply POLICY.md:13]: Wow this is very very well written. And honestly I totally respect what you are saying here, and side with you in agreeance.

Guess my brain has left out all the larger decisions we made because it was obvious, or because it was the only right one. Forgetting that these guidelines went against that.

I'll go ahead and add this in, and likely may include some of what you've said here as well.

Appreciate your input, and seeing this from another point of view!

Comment (confused-Techie) [reply POLICY.md:13]: @DeeDeeG changes have been made as requested

Comment (DeeDeeG) [reply POLICY.md:13]: Are the commits pushed to the PR branch? I can't find it on here.

Comment (confused-Techie) [reply POLICY.md]: @DeeDeeG Sorry, they were in my local stash my bad

File: POLICY.md [hunk 14]
```diff
@@ -8,9 +8,10 @@ A cornerstone of the way the Pulsar team interacts with the community is via Pol
...
-* If a poll fails, then you must not recreate the exact same poll right afterwards. If the same idea has new merit or new validation at a later time, communicate with the community to determine if another poll is reasonable.
+* Whenever we need to make a decision that will affect a large percentage of users/contributors, make a substantial enough change, drastically affect a small percentage of users/contributors then a poll should be made.
+* A Poll should be available for voting for at least 24 hours to accommodate different timezones and work schedules. Polls may be longer than this, but should never be shorter.
+* If a poll fails, then you should not recreate the exact same poll right afterwards. If the same idea has new merit or new validation at a later time, communicate with the community to determine if another poll is reasonable.
+* In some circumstances a poll that meets these criteria will not be made, this is generally the result of the decision being the only viable way to move forward, or if there is already a good enough consensus of the Admin Team.
```
Comment (Daeraxa) [inline POLICY.md:14]:
```suggestion
* In some circumstances a poll that meets these criteria will not be made, this is generally the result of the decision being the only viable way to move forward, or if there is already a good enough consensus of the Pulsar Team.
```
I would change this to Pulsar team just for the sake of consistency

**Comment by confused-Techie** [comment]
Alright, do we think this one looks good?

**Review by DeeDeeG — APPROVED** [review]
Seems to mostly document stuff that is already implicitly or unofficially known amongst core admins, things that community members would have likely seen going on, but it was never written down anywhere particularly before. So, I guess it's good to have somewhere where it's written down! 👍 

---

I can't tell if I'm overthinking this, so putting it below the line:

I'm not sure if it's worth adding something that explains this isn't something people should post as a gotcha. "These are guidelines based on what works for Pulsar, and may not be followed to a T, but they are general practices that arose naturally over time that we tend to at least loosely adhere to. There are no express consequences for failure to follow this policy at this time, but their principles are meant to be good enough to _want_ to adhere to." Something like that??

I kinda feel like you covered that at the top, though.

> This document lists the various policies that have come about naturally as the project progressed and so deserve to be properly documented and viewable by anyone.

I don't want someone trying to bonk us for "not following our own policies!!1" we are out here doing our best. This is about making it so people are aware of how we tend to do things, not so much an accountability thing stemming from the document, rather we are accountable to good principles and maintaining trust by our actions IMO.

File: POLICY.md [hunk 15]
```diff
@@ -0,0 +1,41 @@
+# Policy
+
+This document will lay out the official policies used within Pulsar-Edit. This document lists the various policies that have come about naturally as the project progressed and so deserve to be properly documented and viewable by anyone.
```
Comment (DeeDeeG) [inline POLICY.md:3]: Maybe I am over-thinking this, but I feel like the word "official" here may lend itself to people wanting to bonk us for "not following our own rules!!1"

Maybe delete the first sentence "This document will lay out the official policies [ . . . ]"  in favor of just the second one "This document lists the various policies that have come about naturally [ . . . ]".

These things evolve, I'd rather be accountable to people than documents.

Comment (confused-Techie) [reply POLICY.md:3]: I've just seen this, and while don't have the time to resolve it. Just wanted to say totally agree with the point your making, and I think for this at least it's a good idea to distance ourselves from "official"

Comment (confused-Techie) [reply POLICY.md]: @DeeDeeG Added your changes, since they seemed great to add. Let me know what you think and if we are good to merge. (If you've got the chance, otherwise I can take the previous approval and go ahead)

Comment (DeeDeeG) [reply POLICY.md]: Yeah, the latest changes LGTM, and I noticed a couple typo fixes in there too, good spotting those.

👍

Comment (confused-Techie) [reply POLICY.md]: Thanks for checking through it again! I'll go ahead and merge this one then
